### PR TITLE
VDI Service API Design First Draft

### PR DESCRIPTION
**Pull request by Foxcapades**
First draft of the VDI Service HTTP API.

Rendered HTML doc: https://veupathdb.github.io/service-user-datasets/vdi-api.html

### CODE REVIEW

File: service-vdi/schema/by-path/user-datasets/get.raml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        required: false
+      description:
+        displayName: Dataset Description
+        description: User provided description of the dataset.
+        type: string
+        required: false
+      projectIDs:
+        displayName: Project IDs
+        description: |
+          Project IDs for projects the user dataset record was submitted to.
```
Comment (dmgaldi) [inline service-vdi/schema/by-path/user-datasets/get.raml:44]: Minor: "user dataset" verbage

File: service-vdi/schema/by-path/user-datasets/get.raml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+#%RAML 1.0 Library
```
Comment (dmgaldi) [inline service-vdi/schema/by-path/user-datasets/get.raml:1]: `/by-path/user-datasets/` in directory structure can be updated

Comment (Foxcapades) [reply service-vdi/schema/by-path/user-datasets/get.raml]: renamed

**Review by steve-fischer-200 — APPROVED** [review]
for the paging use num_records instead of limit, to be compatible with wdk
for the import status enum, use:  awaiting, in-progress, done, failed .  this is parallel with the install statuses
change /rpc to /admin
for install clean up, maybe lose the all flag.  if targets array is empty, that means all?
for delete-cleanup, make the time interval (24 hours) be a config property of the service?
in share urls, change path segment {user-id} to {recipient-user-id}
a big one:  the design doc, i believe, calls for an invalid status for an install.  this is different than the old system.  it allows us to do additional validation of an install, based on information we glean from the database, etc

**Review by dmgaldi — APPROVED** [review]

**Pull request merged**